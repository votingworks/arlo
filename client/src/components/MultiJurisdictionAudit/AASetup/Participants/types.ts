import { IAuditSettings } from '../../../../types'

export interface IValues {
  state: IAuditSettings['state']
}
