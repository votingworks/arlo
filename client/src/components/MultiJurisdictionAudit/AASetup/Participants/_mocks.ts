import { readFileSync } from 'fs'
import { join } from 'path'

const jurisdictionFile = new File(
  [
    readFileSync(
      join(
        __dirname,
        '../../../../../public/sample_jurisdiction_filesheet.csv'
      ),
      'utf8'
    ),
  ],
  'jurisdictions.csv',
  { type: 'text/csv' }
)

export default jurisdictionFile
