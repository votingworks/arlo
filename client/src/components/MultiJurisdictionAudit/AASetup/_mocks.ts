import { ElementType } from '../../../types'
import { setupStages, stageTitles } from './index'
import { ISidebarMenuItem } from '../../Atoms/Sidebar'

const relativeStages = (
  stage: ElementType<typeof setupStages>,
  state: ISidebarMenuItem['state'] = 'live'
): {
  prevStage: ISidebarMenuItem
  nextStage: ISidebarMenuItem
  menuItems: ISidebarMenuItem[]
} => {
  const prevId = setupStages[setupStages.indexOf(stage) - 1]
  const prevStage: ISidebarMenuItem = {
    id: prevId,
    title: stageTitles[prevId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const nextId = setupStages[setupStages.indexOf(stage) + 1]
  const nextStage: ISidebarMenuItem = {
    id: nextId,
    title: stageTitles[nextId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const menuItems = setupStages.map((s: ElementType<typeof setupStages>) => ({
    id: s,
    title: stageTitles[s],
    active: s === stage,
    activate: jest.fn(),
    state,
  }))
  return { prevStage, nextStage, menuItems }
}

export default relativeStages
