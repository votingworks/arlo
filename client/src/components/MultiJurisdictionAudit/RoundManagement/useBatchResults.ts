import { useEffect, useState } from 'react'
import { toast } from 'react-toastify'
import { api } from '../../utilities'

export interface IResultValues {
  [batchId: string]: {
    [choiceId: string]: number | string
  }
}

export interface IBatch {
  id: string
  name: string
  numBallots: number
  auditBoard: null | {
    id: string
    name: string
  }
}

const stringifyPossibleNull = (v: string | number | null) => (v ? `${v}` : '')

const reformatResults = (r: IResultValues, numberify = true): IResultValues => {
  return Object.keys(r).reduce(
    (a, batchId) => ({
      ...a,
      [batchId]: Object.keys(r[batchId]).reduce(
        (b, choiceId) => ({
          ...b,
          [choiceId]: numberify
            ? Number(r[batchId][choiceId])
            : stringifyPossibleNull(r[batchId][choiceId]),
        }),
        {}
      ),
    }),
    {}
  )
}

const numberifyResults = (r: IResultValues): IResultValues => reformatResults(r)

const getResults = async (
  electionId: string,
  jurisdictionId: string,
  roundId: string
): Promise<IResultValues | null> => {
  const response = await api<IResultValues>(
    `/election/${electionId}/jurisdiction/${jurisdictionId}/round/${roundId}/batches/results`
  )
  if (!response) return null
  return reformatResults(response, false)
}

const getBatches = async (
  electionId: string,
  jurisdictionId: string,
  roundId: string
): Promise<IBatch[] | null> => {
  const response = await api<{ batches: IBatch[] }>(
    `/election/${electionId}/jurisdiction/${jurisdictionId}/round/${roundId}/batches`
  )
  if (!response) return null
  return response.batches
}

const useBatchResults = (
  electionId: string,
  jurisdictionId: string,
  roundId: string
): [
  IResultValues | null,
  IBatch[] | null,
  (arg0: IResultValues) => Promise<boolean>
] => {
  const [results, setResults] = useState<IResultValues | null>(null)
  const [batches, setBatches] = useState<IBatch[] | null>(null)

  const updateResults = async (newResults: IResultValues): Promise<boolean> => {
    try {
      await api(
        `/election/${electionId}/jurisdiction/${jurisdictionId}/round/${roundId}/batches/results`,
        {
          method: 'PUT',
          // stringify and numberify the contests (all number values are handled as strings clientside, but are required as numbers serverside)
          body: JSON.stringify(numberifyResults(newResults)),
          headers: {
            'Content-Type': 'application/json',
          },
        }
      )
    } catch (err) /* istanbul ignore next */ {
      // TODO move toasting into api
      toast.error(err.message)
      return false
    }
    setResults(newResults)
    return true
  }

  useEffect(() => {
    ;(async () => {
      const newBatches = await getBatches(electionId, jurisdictionId, roundId)
      const newResults = await getResults(electionId, jurisdictionId, roundId)
      setBatches(newBatches)
      setResults(newResults)
    })()
  }, [electionId, jurisdictionId, roundId])
  return [results, batches, updateResults]
}

export default useBatchResults
