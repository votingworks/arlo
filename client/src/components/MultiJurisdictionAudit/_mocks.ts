import { contestMocks } from './AASetup/Contests/_mocks'
import { IFileInfo } from './useJurisdictions'
import {
  auditSettings,
  manifestFile,
  talliesFile,
} from './useSetupMenuItems/_mocks'
import { FileProcessingStatus } from './useSetupMenuItems/getJurisdictionFileStatus'
import { IAuditSettings } from '../../types'
import { IBallot } from './RoundManagement/useBallots'
import { IAuditBoard } from './useAuditBoards'
import { IRound } from './useRoundsAuditAdmin'

const manifestFormData: FormData = new FormData()
manifestFormData.append('manifest', manifestFile, manifestFile.name)
const talliesFormData: FormData = new FormData()
talliesFormData.append('batchTallies', talliesFile, talliesFile.name)

export const jaApiCalls = {
  getUser: {
    url: '/api/me',
    response: {
      type: 'jurisdiction_admin',
      name: 'Joe',
      email: 'test@email.org',
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          election: {
            id: '1',
            auditName: 'audit one',
            electionName: 'election one',
            state: 'AL',
          },
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          election: {
            id: '2',
            auditName: 'audit two',
            electionName: 'election two',
            state: 'AL',
          },
        },
        {
          id: 'jurisdiction-id-3',
          name: 'Jurisdiction Three',
          election: {
            id: '1',
            auditName: 'audit one',
            electionName: 'election one',
            state: 'AL',
          },
        },
      ],
      organizations: [],
    },
  },
  getRounds: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/round',
    response: { rounds: [] },
  },
  getBallotManifestFile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/ballot-manifest',
    response,
  }),
  getBatchTalliesFile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/batch-tallies',
    response,
  }),
  getCVRSfile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/cvrs',
    response,
  }),
  getSettings: (response: IAuditSettings) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/settings',
    response,
  }),
  putManifest: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/ballot-manifest',
    options: {
      method: 'PUT',
      body: manifestFormData,
    },
    response: { status: 'ok' },
  },
  putTallies: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/batch-tallies',
    options: {
      method: 'PUT',
      body: talliesFormData,
    },
    response: { status: 'ok' },
  },
  getAuditBoards: (auditBoards: IAuditBoard[]) => ({
    url:
      '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/audit-board',
    response: { auditBoards },
  }),
  getBallots: (ballots: IBallot[]) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/ballots',
    response: { ballots },
  }),
}

export const aaApiCalls = {
  getUser: {
    url: '/api/me',
    response: {
      type: 'audit_admin',
      name: 'Joe',
      email: 'test@email.org',
      jurisdictions: [],
      organizations: [
        {
          id: 'org-id',
          name: 'State of California',
          elections: [],
        },
      ],
    },
  },
  getRounds: (rounds: IRound[]) => ({
    url: '/api/election/1/round',
    response: { rounds },
  }),
  getJurisdictions: {
    url: '/api/election/1/jurisdiction',
    response: {
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          ballotManifest: { file: null, processing: null },
          currentRoundStatus: null,
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          ballotManifest: { file: null, processing: null },
          currentRoundStatus: null,
        },
      ],
    },
  },
  getBatchJurisdictions: {
    url: '/api/election/1/jurisdiction',
    response: {
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          ballotManifest: { file: null, processing: null },
          batchTallies: { file: null, processing: null },
          currentRoundStatus: null,
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          ballotManifest: { file: null, processing: null },
          batchTallies: { file: null, processing: null },
          currentRoundStatus: null,
        },
      ],
    },
  },
  getJurisdictionFile: {
    url: '/api/election/1/jurisdiction/file',
    response: {
      file: {
        contents: null,
        name: 'file name',
        uploadedAt: 'a long time ago in a galaxy far far away',
      },
      processing: {
        status: FileProcessingStatus.Processed,
        error: null,
        startedAt: 'once upon a time',
        endedAt: 'and they lived happily ever after',
      },
    },
  },
  getContests: {
    url: '/api/election/1/contest',
    response: contestMocks.filledTargeted,
  },
  getSettings: (response: IAuditSettings) => ({
    url: '/api/election/1/settings',
    response,
  }),
  putSettings: {
    url: '/api/election/1/settings',
    options: {
      method: 'PUT',
      body: JSON.stringify(auditSettings.all),
      headers: { 'Content-Type': 'application/json' },
    },
    response: { status: 'ok' },
  },
  getSampleSizes: {
    url: '/api/election/1/sample-sizes',
    response: { sampleSizes: null },
  },
}
