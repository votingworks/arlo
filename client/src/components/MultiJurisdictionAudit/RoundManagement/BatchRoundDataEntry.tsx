import React from 'react'
import { useParams } from 'react-router-dom'
import { H5 } from '@blueprintjs/core'
import { Field, Formik, Form, FormikProps } from 'formik'
import styled from 'styled-components'
import useContestsJurisdictionAdmin from './useContestsJurisdictionAdmin'
import Card from '../../Atoms/SpacedCard'
import FormField from '../../Atoms/Form/FormField'
import FormButton from '../../Atoms/Form/FormButton'
import { testNumber } from '../../utilities'
import useBatchResults, { IResultValues } from './useBatchResults'
import { IRound } from '../useRoundsAuditAdmin'

const BottomButton = styled(FormButton)`
  margin: 30px 0;
`

const BlockLabel = styled.label`
  display: block;
  margin: 20px 0;
`

interface IProps {
  round: IRound
}

interface IValues {
  results: IResultValues
}

const BatchRoundDataEntry = ({ round }: IProps) => {
  const { electionId, jurisdictionId } = useParams<{
    electionId: string
    jurisdictionId: string
  }>()
  const allContests = useContestsJurisdictionAdmin(electionId, jurisdictionId)
  const [results, batches, updateResults] = useBatchResults(
    electionId,
    jurisdictionId,
    round.id
  )

  if (!results || !allContests || !batches) return null

  // batch comparison audits only support a single contest for now
  const contest = allContests[0]

  const alreadySubmittedResults = Object.values(results).some(a =>
    Object.values(a).some(b => b)
  )

  const submit = async (values: IValues) => {
    updateResults(values.results)
  }

  return (
    <Formik initialValues={{ results }} enableReinitialize onSubmit={submit}>
      {({ handleSubmit }: FormikProps<IValues>) => (
        <Form>
          <p>
            When you have examined all the ballots assigned to you, enter the
            number of votes recorded for each candidate/choice from the audited
            ballots.
          </p>
          {batches.map(batch => (
            <Card key={batch.id}>
              <H5>{`Batch: ${batch.name}, Contest: ${contest.name}`}</H5>
              {contest.choices.map(choice => (
                <BlockLabel
                  key={choice.id}
                  htmlFor={`results[${batch.id}][${choice.id}]`}
                >
                  Votes for {choice.name}:
                  {alreadySubmittedResults ? (
                    results[batch.id][choice.id]
                  ) : (
                    <Field
                      id={`results[${batch.id}][${choice.id}]`}
                      name={`results[${batch.id}][${choice.id}]`}
                      disabled={alreadySubmittedResults}
                      validate={testNumber()}
                      component={FormField}
                    />
                  )}
                </BlockLabel>
              ))}
            </Card>
          ))}
          <BottomButton
            type="submit"
            intent="primary"
            disabled={alreadySubmittedResults}
            onClick={handleSubmit}
          >
            {alreadySubmittedResults
              ? `Already Submitted Data for Round ${round.roundNum}`
              : `Submit Data for Round ${round.roundNum}`}
          </BottomButton>
        </Form>
      )}
    </Formik>
  )
}

export default BatchRoundDataEntry
