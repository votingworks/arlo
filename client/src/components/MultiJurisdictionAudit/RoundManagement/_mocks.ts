import { IBatch } from './useBatchResults'

export interface INullResultValues {
  [contestId: string]: {
    [choiceId: string]: null | string | number
  }
}

export const resultsMocks: {
  [key in 'emptyInitial' | 'complete']: INullResultValues
} = {
  emptyInitial: {
    'contest-id-1': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'contest-id-2': {
      'choice-id-3': null,
      'choice-id-4': null,
    },
  },
  complete: {
    'contest-id-1': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'contest-id-2': {
      'choice-id-3': 1,
      'choice-id-4': 2,
    },
  },
}

export const batchResultsMocks: {
  [key in 'empty' | 'complete']: INullResultValues
} = {
  empty: {
    'batch-1': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'batch-2': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'batch-3': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
  },
  complete: {
    'batch-1': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'batch-2': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'batch-3': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
  },
}

export const batchesMocks: {
  [key in 'emptyInitial' | 'complete']: { batches: IBatch[] }
} = {
  emptyInitial: {
    batches: [
      {
        id: 'batch-1',
        name: 'Batch One',
        numBallots: 100,
        auditBoard: null,
      },
      {
        id: 'batch-2',
        name: 'Batch Two',
        numBallots: 100,
        auditBoard: null,
      },
      {
        id: 'batch-3',
        name: 'Batch Three',
        numBallots: 100,
        auditBoard: null,
      },
    ],
  },
  complete: {
    batches: [
      {
        id: 'batch-1',
        name: 'Batch One',
        numBallots: 100,
        auditBoard: {
          id: 'ab-1',
          name: 'Audit Board One',
        },
      },
      {
        id: 'batch-2',
        name: 'Batch Two',
        numBallots: 100,
        auditBoard: {
          id: 'ab-1',
          name: 'Audit Board One',
        },
      },
      {
        id: 'batch-3',
        name: 'Batch Three',
        numBallots: 100,
        auditBoard: {
          id: 'ab-1',
          name: 'Audit Board One',
        },
      },
    ],
  },
}

export default resultsMocks
