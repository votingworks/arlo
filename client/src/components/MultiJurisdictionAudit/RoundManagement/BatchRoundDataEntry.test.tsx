import React from 'react'
import { screen, fireEvent } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { useParams } from 'react-router-dom'
import BatchRoundDataEntry from './BatchRoundDataEntry'
import { roundMocks, contestMocks } from '../useSetupMenuItems/_mocks'
import { batchesMocks, batchResultsMocks, INullResultValues } from './_mocks'
import { renderWithRouter, withMockFetch } from '../../testUtilities'
import { IContest } from '../../../types'
import { IBatch } from './useBatchResults'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useRouteMatch: jest.fn(),
  useParams: jest.fn(),
}))
const paramsMock = useParams as jest.Mock
paramsMock.mockReturnValue({
  electionId: '1',
  jurisdictionId: '1',
})

const apiCalls = {
  getJAContests: (response: { contests: IContest[] }) => ({
    url: `/api/election/1/jurisdiction/1/contest`,
    response,
  }),
  getBatches: (response: { batches: IBatch[] }) => ({
    url: '/api/election/1/jurisdiction/1/round/round-1/batches',
    response,
  }),
  getResults: (response: INullResultValues) => ({
    url: '/api/election/1/jurisdiction/1/round/round-1/batches/results',
    response,
  }),
  putResults: (results: INullResultValues) => ({
    url: '/api/election/1/jurisdiction/1/round/round-1/batches/results',
    options: {
      method: 'PUT',
      body: JSON.stringify(results),
      headers: {
        'Content-Type': 'application/json',
      },
    },
    response: { status: 'ok' },
  }),
}

describe('offline round data entry', () => {
  it('renders', async () => {
    const expectedCalls = [
      apiCalls.getJAContests({ contests: contestMocks.oneTargeted }),
      apiCalls.getBatches(batchesMocks.emptyInitial),
      apiCalls.getResults(batchResultsMocks.empty),
    ]
    await withMockFetch(expectedCalls, async () => {
      const { container } = renderWithRouter(
        <BatchRoundDataEntry round={roundMocks.singleIncomplete[0]} />,
        {
          route: '/election/1/jurisdiction/1',
        }
      )
      await screen.findByText('Round 1 Data Entry')
      expect(container).toMatchSnapshot()
    })
  })

  it('renders after submission', async () => {
    const expectedCalls = [
      apiCalls.getJAContests({ contests: contestMocks.oneTargeted }),
      apiCalls.getBatches(batchesMocks.complete),
      apiCalls.getResults(batchResultsMocks.complete),
    ]
    await withMockFetch(expectedCalls, async () => {
      const { container } = renderWithRouter(
        <BatchRoundDataEntry round={roundMocks.singleIncomplete[0]} />,
        {
          route: '/election/1/jurisdiction/1',
        }
      )
      await screen.findByText('Round 1 Data Entry')
      expect(container).toMatchSnapshot()
    })
  })

  it('submits', async () => {
    const expectedCalls = [
      apiCalls.getJAContests({ contests: contestMocks.oneTargeted }),
      apiCalls.getBatches(batchesMocks.emptyInitial),
      apiCalls.getResults(batchResultsMocks.empty),
      apiCalls.putResults(batchResultsMocks.complete),
    ]
    await withMockFetch(expectedCalls, async () => {
      const { container } = renderWithRouter(
        <BatchRoundDataEntry round={roundMocks.singleIncomplete[0]} />,
        {
          route: '/election/1/jurisdiction/1',
        }
      )
      await screen.findByText('Round 1 Data Entry')
      ;[0, 1, 2].forEach(batch => {
        fireEvent.change(
          screen.getAllByLabelText('Votes for Choice One:')[batch],
          {
            target: { value: '1' },
          }
        )
        fireEvent.change(
          screen.getAllByLabelText('Votes for Choice Two:')[batch],
          {
            target: { value: '2' },
          }
        )
      })
      userEvent.click(screen.getByText('Submit Data for Round 1'))
      await screen.findByText('Already Submitted Data for Round 1')
      expect(container).toMatchSnapshot()
    })
  })
})
