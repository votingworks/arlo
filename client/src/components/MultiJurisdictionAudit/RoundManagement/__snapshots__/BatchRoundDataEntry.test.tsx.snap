// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offline round data entry renders 1`] = `
<div>
  <form>
    <h2
      class="bp3-heading sc-bdVaJa ibVpPq"
    >
      Round 
      1
       Data Entry
    </h2>
    <p>
      When you have examined all the ballots assigned to you, enter the number of votes recorded for each candidate/choice from the audited ballots.
    </p>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Batch: Batch One, Contest: Contest 1
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-1][choice-id-1]"
      >
        Votes for 
        Choice One
        :
        <div
          class="sc-htpNat gdpVCF"
        >
          <div
            class="bp3-input-group sc-bxivhb eSUmfF"
          >
            <input
              class="bp3-input"
              id="results[batch-1][choice-id-1]"
              name="results[batch-1][choice-id-1]"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-1][choice-id-2]"
      >
        Votes for 
        Choice Two
        :
        <div
          class="sc-htpNat gdpVCF"
        >
          <div
            class="bp3-input-group sc-bxivhb eSUmfF"
          >
            <input
              class="bp3-input"
              id="results[batch-1][choice-id-2]"
              name="results[batch-1][choice-id-2]"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
    </div>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Batch: Batch Two, Contest: Contest 1
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-2][choice-id-1]"
      >
        Votes for 
        Choice One
        :
        <div
          class="sc-htpNat gdpVCF"
        >
          <div
            class="bp3-input-group sc-bxivhb eSUmfF"
          >
            <input
              class="bp3-input"
              id="results[batch-2][choice-id-1]"
              name="results[batch-2][choice-id-1]"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-2][choice-id-2]"
      >
        Votes for 
        Choice Two
        :
        <div
          class="sc-htpNat gdpVCF"
        >
          <div
            class="bp3-input-group sc-bxivhb eSUmfF"
          >
            <input
              class="bp3-input"
              id="results[batch-2][choice-id-2]"
              name="results[batch-2][choice-id-2]"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
    </div>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Batch: Batch Three, Contest: Contest 1
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-3][choice-id-1]"
      >
        Votes for 
        Choice One
        :
        <div
          class="sc-htpNat gdpVCF"
        >
          <div
            class="bp3-input-group sc-bxivhb eSUmfF"
          >
            <input
              class="bp3-input"
              id="results[batch-3][choice-id-1]"
              name="results[batch-3][choice-id-1]"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-3][choice-id-2]"
      >
        Votes for 
        Choice Two
        :
        <div
          class="sc-htpNat gdpVCF"
        >
          <div
            class="bp3-input-group sc-bxivhb eSUmfF"
          >
            <input
              class="bp3-input"
              id="results[batch-3][choice-id-2]"
              name="results[batch-3][choice-id-2]"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
    </div>
    <button
      class="bp3-button bp3-intent-primary sc-dnqmqq bytmcb sc-htoDjs dESvso"
      type="submit"
    >
      <span
        class="bp3-button-text"
      >
        Submit Data for Round 1
      </span>
    </button>
  </form>
</div>
`;

exports[`offline round data entry renders after submission 1`] = `
<div>
  <form>
    <h2
      class="bp3-heading sc-bdVaJa ibVpPq"
    >
      Round 
      1
       Data Entry
    </h2>
    <p>
      When you have examined all the ballots assigned to you, enter the number of votes recorded for each candidate/choice from the audited ballots.
    </p>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Batch: Batch One, Contest: Contest 1
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-1][choice-id-1]"
      >
        Votes for 
        Choice One
        :
        1
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-1][choice-id-2]"
      >
        Votes for 
        Choice Two
        :
        2
      </label>
    </div>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Batch: Batch Two, Contest: Contest 1
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-2][choice-id-1]"
      >
        Votes for 
        Choice One
        :
        1
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-2][choice-id-2]"
      >
        Votes for 
        Choice Two
        :
        2
      </label>
    </div>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Batch: Batch Three, Contest: Contest 1
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-3][choice-id-1]"
      >
        Votes for 
        Choice One
        :
        1
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-3][choice-id-2]"
      >
        Votes for 
        Choice Two
        :
        2
      </label>
    </div>
    <button
      class="bp3-button bp3-disabled bp3-intent-primary sc-dnqmqq bytmcb sc-htoDjs dESvso"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      <span
        class="bp3-button-text"
      >
        Already Submitted Data for Round 1
      </span>
    </button>
  </form>
</div>
`;

exports[`offline round data entry submits 1`] = `
<div>
  <form>
    <h2
      class="bp3-heading sc-bdVaJa ibVpPq"
    >
      Round 
      1
       Data Entry
    </h2>
    <p>
      When you have examined all the ballots assigned to you, enter the number of votes recorded for each candidate/choice from the audited ballots.
    </p>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Batch: Batch One, Contest: Contest 1
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-1][choice-id-1]"
      >
        Votes for 
        Choice One
        :
        1
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-1][choice-id-2]"
      >
        Votes for 
        Choice Two
        :
        2
      </label>
    </div>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Batch: Batch Two, Contest: Contest 1
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-2][choice-id-1]"
      >
        Votes for 
        Choice One
        :
        1
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-2][choice-id-2]"
      >
        Votes for 
        Choice Two
        :
        2
      </label>
    </div>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Batch: Batch Three, Contest: Contest 1
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-3][choice-id-1]"
      >
        Votes for 
        Choice One
        :
        1
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[batch-3][choice-id-2]"
      >
        Votes for 
        Choice Two
        :
        2
      </label>
    </div>
    <button
      class="bp3-button bp3-disabled bp3-intent-primary sc-dnqmqq bytmcb sc-htoDjs dESvso"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      <span
        class="bp3-button-text"
      >
        Already Submitted Data for Round 1
      </span>
    </button>
  </form>
</div>
`;
